wolfcrypt/src/port/iotsafe/iotsafe.c

```diff
@@ -1237,16 +1414,23 @@ int wolfSSL_CTX_iotsafe_enable(WOLFSSL_CTX *ctx)
     return 0;
 }
 
-int wolfSSL_iotsafe_on(WOLFSSL *ssl, byte privkey_id,
-       byte ecdh_keypair_slot, byte peer_pubkey_slot, byte peer_cert_slot)
+int wolfSSL_iotsafe_on_ex(WOLFSSL *ssl, byte *privkey_id,
+       byte *ecdh_keypair_slot, byte *peer_pubkey_slot, byte *peer_cert_slot, word16 id_size)
 {
 #if defined(HAVE_PK_CALLBACKS)
     int ret;
     IOTSAFE iotsafe;
-    iotsafe.privkey_id = privkey_id;
-    iotsafe.ecdh_keypair_slot = ecdh_keypair_slot;
-    iotsafe.peer_pubkey_slot = peer_pubkey_slot;
-    iotsafe.peer_cert_slot = peer_cert_slot;
+#ifdef TWO_BYTES_ID_DEMO
```

Comment:
Why not always use `id_size` and XMEMCPY or sizeof() the dest? If required you could force set id_size = 2 when `TWO_BYTES_ID_DEMO` is defined. Maybe it would be better to have this define `IOTSAFE_ID_SIZE = 2`? Have you considered the big endian cases?

Reply:
You are definitely right. `_DEMO` defines do not belong here, and the XMEMCPY case is generic enough to cope with id_size = 1 on itself. Ifdef removed.